cmd/system-probe/subcommands/usm/sysinfo_linux.go

```diff
@@ -195,3 +261,75 @@ func formatCmdline(args []string) string {
 	}
 	return builder.String()
 }
+
+// extractDDServiceFromProc reads DD_SERVICE from /proc/PID/environ using the service discovery infrastructure
+// This mirrors the approach from pkg/collector/corechecks/servicediscovery/module/envs.go
```

Comment:
Why do we need to copy? Can't we just expose the original logic and import it?

Reply:
I wasn’t sure whether I wanted to involve service discovery in this change, but I can do it, no problem.